compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -279,7 +279,7 @@ impl DefinitionId {
 
 impl From<DefinitionId> for Index {
     fn from(id: DefinitionId) -> Self {
-        Index::from_raw_parts(id.0, u64::MAX)
```

Comment:
The reason DefinitionIds never clashed before was this impl used `u64::MAX` for the generational index so it was never equal to the indices for any expression node since those always used `0` for the generational index.